### PR TITLE
🐛fix xref transform bug

### PR DESCRIPTION
**Pull request by OsmondJiang**
xref transfrom fallback logic:

resolvedHref -> data-raw-html->data-raw-source -> <span class="xref">xref-value</span>

fix: https://dev.azure.com/ceapex/Engineering/_workitems/edit/105007

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/4887)

### CODE REVIEW

**Comment by OsmondJiang** [comment]
@yufeih  Integration test sometimes failed with below exception:

image

**Comment by OsmondJiang** [comment]
That's what I got from test PR:

image

**Comment by yufeih** [comment]
@fenxu , log in Docs.Build does not flush on dispose?

**Comment by OsmondJiang** [comment]
> @fenxu , log in Docs.Build does not flush on dispose?

It's related to concurrency append lines to same file, has created a PR to fix it.

https://dev.azure.com/ceapex/Engineering/_git/Docs.Build/pullrequest/13676?_a=files&path=%2Fsrc%2Fshared%2Flogger%2FLogger.cs

File: src/docfx/lib/HtmlUtility.cs [hunk 1]
```diff
@@ -152,7 +152,7 @@ public static string TransformXref(string html, Func<string, bool, int, (string
                 var xref = HttpUtility.HtmlDecode(node.GetAttributeValue("href", ""));
 
                 var raw = HttpUtility.HtmlDecode(
-                    node.GetAttributeValue("data-raw-html", null) ?? node.GetAttributeValue("data-raw-source", null) ?? "");
+                    node.GetAttributeValue("data-raw-html", null) ?? node.GetAttributeValue("data-raw-source", null) ?? $"<span class=\"xref\">{UrlUtility.SplitUrl(xref).path}</span>");
```
Comment (yufeih) [inline src/docfx/lib/HtmlUtility.cs]: Also share this logic with `XrefExtension.cs`, line 27, 37 ?

Comment (OsmondJiang) [reply src/docfx/lib/HtmlUtility.cs]: @live1206 , do we need share? seems the logic is not totally same

Comment (OsmondJiang) [reply src/docfx/lib/HtmlUtility.cs]: We don't need to, the "data-raw-source" generated from markdig here is always itself, so we don't need add fallback logic there.

**Comment by fenxu** [comment]
The log issue is fixed, not flush but concurrent. Some code is removed. Anyway, it works now.

File: src/docfx/lib/HtmlUtility.cs (hunk 1 above)
Comment (yufeih) [reply src/docfx/lib/HtmlUtility.cs]: I see, should HTML encode the content

**Pull request merged**